jsettlers.logic/src/main/java/jsettlers/ai/economy/BuildingListEconomyMinister.java

```diff
@@ -0,0 +1,416 @@
+		}
+	}
+
+	private boolean isHighGoodsGame(AiStatistics aiStatistics, Player player) {
+		byte playerId = player.playerId;
+		return aiStatistics.getNumberOfMaterialTypeForPlayer(EMaterialType.AXE, playerId) >= 8 &&
+				aiStatistics.getNumberOfMaterialTypeForPlayer(EMaterialType.SAW, playerId) >= 3 &&
+				aiStatistics.getNumberOfMaterialTypeForPlayer(EMaterialType.PICK, playerId) >= 5;
+	}
+
+	private boolean isMiddleGoodsGame(AiStatistics aiStatistics, Player player) {
+		byte playerId = player.playerId;
+		return aiStatistics.getNumberOfMaterialTypeForPlayer(EMaterialType.AXE, playerId) >= 6 &&
+				aiStatistics.getNumberOfMaterialTypeForPlayer(EMaterialType.SAW, playerId) >= 2 &&
+				aiStatistics.getNumberOfMaterialTypeForPlayer(EMaterialType.PICK, playerId) >= 4;
+	}
+
+	private void addGoldBuildings(AiMapInformation aiMapInformation, List<EBuildingType> weaponsBuildings) {
+		if (aiMapInformation.getNumberOfGoldMelts() > 0) {
+			weaponsBuildings.add(GOLDMINE);
+			for (int ii = 0; ii < aiMapInformation.getNumberOfGoldMelts(); ii++) {
+				weaponsBuildings.add(GOLDMELT);
+				weaponsBuildings.add(STOCK);
+			}
+		}
+	}
+
+	@Override
+	public int getNumberOfParallelConstructionSides(AiStatistics aiStatistics, byte playerId) {
+		if (aiStatistics.getNumberOfMaterialTypeForPlayer(EMaterialType.PLANK, playerId) > 1
+				&& aiStatistics.getNumberOfMaterialTypeForPlayer(EMaterialType.STONE, playerId) > 1) {
+			// If plank and stone is still offered, we can build the next building.
+			// If the next building will consume all remaining offers we won't return 100 in the next tick
+			return 100;
+		}
+		return Math.max((int) Math.ceil((float) aiStatistics.getNumberOfBuildingTypeForPlayer(LUMBERJACK, playerId) / 2F), 2);
+	}
+
+	@Override
+	public List<EBuildingType> getBuildingsToBuild(AiStatistics aiStatistics, byte playerId) {
+		List<EBuildingType> allBuildingsToBuild = getRushDefenceBuildingsIfWeAreInDanger(aiStatistics, playerId);
+		allBuildingsToBuild.addAll(buildingsToBuild);
+		return allBuildingsToBuild;
+	}
+
+	private List<EBuildingType> getRushDefenceBuildingsIfWeAreInDanger(AiStatistics aiStatistics, byte playerId) {
+		if (aiStatistics.getTotalNumberOfBuildingTypeForPlayer(BARRACK, playerId) < 1) {
+			for (byte enemy : aiStatistics.getEnemiesOf(playerId)) {
+				if (aiStatistics.getNumberOfBuildingTypeForPlayer(WEAPONSMITH, enemy) > 0) {
+					List<EBuildingType> defenceBuildings = new ArrayList<>();
+					defenceBuildings.add(LUMBERJACK);
+					defenceBuildings.add(SAWMILL);
+					defenceBuildings.add(STONECUTTER);
+					defenceBuildings.add(IRONMELT);
+					defenceBuildings.add(WEAPONSMITH);
+					defenceBuildings.add(BARRACK);
+					defenceBuildings.add(SMALL_LIVINGHOUSE);
+					defenceBuildings.add(COALMINE);
+					defenceBuildings.add(IRONMINE);
+					defenceBuildings.add(MEDIUM_LIVINGHOUSE);
+					return defenceBuildings;
+				}
+			}
+		}
+		// We are not in danger. We need no defence.
+		return new ArrayList<>();
```

Comment:
Does this list need to be modified later?
If not:
- Extract the first one to be a global List. Use `Arrays.asList()` for it.
- Use `Collections.emptyList()` here.